python/paddle/v2/framework/optimizer.py

```diff
@@ -272,3 +272,60 @@ def _append_optimize_op(self, block, param_and_grad):
             attrs={"mu": self._momentum})
 
         return momentum_op
+
+
+class AdagradOptimizer(Optimizer):
+    """Simple Adagrad optimizer with moment state
+    """
+    _moment_acc_str = "moment"
+
+    def __init__(self, learning_rate, epsilon):
```

Comment:
You can give a default value to the epsilon.

Reply:
Also if the original paper mentions a default learning rate value for the Adagrad optimizer, then you can add it too.

Reply:
Done. The original paper does not specify a default learning rate. Only added default value to epsilon.